File: core/src/Dignite.Paperbase.Application.Contracts/Permissions/PaperbasePermissions.cs
using Volo.Abp.Reflection;

namespace Dignite.Paperbase.Permissions;

public class PaperbasePermissions
{
    public const string GroupName = "Paperbase";

    public static class Documents
    {
        public const string Default = GroupName + ".Documents";
        public const string Upload = Default + ".Upload";
        public const string Delete = Default + ".Delete";
        public const string PermanentDelete = Default + ".PermanentDelete";
        public const string Restore = Default + ".Restore";
        public const string Export = Default + ".Export";
        public const string ConfirmClassification = Default + ".ConfirmClassification";

        public static class Pipelines
        {
            public const string Default = Documents.Default + ".Pipelines";
            public const string Retry = Default + ".Retry";
        }
    }

    public static class Chat
    {
        public const string Default = GroupName + ".Chat";
        public const string Create = Default + ".Create";
        public const string SendMessage = Default + ".SendMessage";
        public const string Delete = Default + ".Delete";
    }

    public static class DocumentRelations
    {
        public const string Default = GroupName + ".DocumentRelations";
        public const string Create = Default + ".Create";
        public const string Delete = Default + ".Delete";
        public const string ConfirmRelation = Default + ".ConfirmRelation";
    }

    public static string[] GetAll()
    {
        return ReflectionHelper.GetPublicConstantsRecursively(typeof(PaperbasePermissions));
    }
}

File: angular/packages/paperbase/src/lib/shared/tokens/paperbase-permissions.ts
export const PAPERBASE_PERMISSIONS = {
  Documents: {
    Default: 'Paperbase.Documents',
    Upload: 'Paperbase.Documents.Upload',
    Delete: 'Paperbase.Documents.Delete',
    PermanentDelete: 'Paperbase.Documents.PermanentDelete',
    Restore: 'Paperbase.Documents.Restore',
    Export: 'Paperbase.Documents.Export',
    ConfirmClassification: 'Paperbase.Documents.ConfirmClassification',
    Pipelines: {
      Default: 'Paperbase.Documents.Pipelines',
      Retry: 'Paperbase.Documents.Pipelines.Retry',
    },
    Templates: {
      Default: 'Paperbase.Documents.Templates',
      Create: 'Paperbase.Documents.Templates.Create',
      Update: 'Paperbase.Documents.Templates.Update',
      Delete: 'Paperbase.Documents.Templates.Delete',
    },
  },
  Cabinets: {
    Default: 'Paperbase.Cabinets',
    Create: 'Paperbase.Cabinets.Create',
    Update: 'Paperbase.Cabinets.Update',
    Delete: 'Paperbase.Cabinets.Delete',
  },
} as const;
